src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -839,59 +854,194 @@ mergeRuns resolve hfs hbio caching alloc runPaths mergeLevel runs = do
   Credits
 -------------------------------------------------------------------------------}
 
-type Credit = Int
+{-
+  Note [Credits]
```

Comment:
Should also note that it is okay if a merge finishes earlier than expected, as long as it does not finish *too* early